### PR TITLE
Get rustc commit hash from channel manifest

### PR DESCRIPTION
**Pull request by SimonSapin**
… added in https://github.com/rust-lang/rust/pull/44218, instead of using the GitHub API.

Also upgrade to rustc 1.22.0-nightly (d93036a04 2017-09-07).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18420)

### CODE REVIEW

**Comment by SimonSapin** [comment]
On buildbot:

```
error: failed to run custom build command for `fontsan v0.3.2 (https://github.com/servo/fontsan#ab68da17)`
could not execute process `/Users/servo/buildbot/slave/mac-dev-unit/build/target/debug/build/fontsan-c0e0814b33162e4e/build-script-build` (never executed)
```

On Travis:

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64"   […]
  = note: collect2: error: ld returned 1 exit status
          
error: aborting due to previous error
error: Could not compile `servo`.
```

… neither with any more output that seems relevant to the error. I can’t reproduce locally, on either Linux or OS X. So… I’ve no idea. Let’s see if it’s reliable?

@bors-servo retry

**Comment by SimonSapin** [comment]
At least some of the failures are:

```
  │ ERROR:servo: assertion failed: distance_to_stack_bottom <= max_allowable_distance
  │ assertion failed: distance_to_stack_bottom <= max_allowable_distance (thread <unnamed>, at /home/servo/buildbot/slave/linux-rel-css/build/components/style/context.rs:658)
```

CC @bholley @julian-seward1, should we increase the stack size?

**Comment by bholley** [comment]
I don't think stack size is the issue - we just bumped the limit, and those assertion stacks are very short in any case.

We need to understand what the various values are to understand why it isn't working.

**Comment by bholley** [comment]
(Note that we could increase the slop from 10k, but if that fixed it, that would also be surprising enough to warrant figuring out why).

**Comment by jdm** [comment]
More of the same assertions.

**Comment by SimonSapin** [comment]
Bisection in #18445 shows that https://github.com/rust-lang/rust/pull/43931 is responsible (though possibly indirectly). Maybe more inlining is making some stack frames larger? @bholley, @emilio, would that make sense?

**Comment by emilio** [comment]
I think so, yeah.

We can confirm it marking `StyleBloom::new` and `StyleSharingCandidateCache::new` with `#[inline(never)]`.

File: components/style/bloom.rs [hunk 1]
```diff
@@ -121,6 +121,7 @@ impl<E: TElement> StyleBloom<E> {
     /// Create an empty `StyleBloom`. Because StyleBloom acquires the thread-
     /// local filter buffer, creating multiple live StyleBloom instances at
     /// the same time on the same thread will panic.
+    #[inline(never)]
```
Comment (emilio) [inline components/style/bloom.rs]: Maybe leave a comment on the reasoning for this here and below?

**Review by emilio — APPROVED** [review]

File: components/style/bloom.rs (hunk 1 above)
Comment (SimonSapin) [reply components/style/bloom.rs]: Done.

**Pull request merged**